invenio_rdm_migrator/streams/actions/load/communities.py

```diff
@@ -0,0 +1,242 @@
+        ("bucket", "id", generate_uuid),
+        ("community_file", "id", generate_uuid),
+    ]
+
+    name = "community-create"
+    data_cls = CommunityData
+
+    def _generate_rows(self, **kwargs):
+        """Generates rows for a new community."""
+        community_id = self.data.community["id"]
+        slug = self.data.community["slug"]
+        community_file_id = None
+        logo_object_version_id = None
+
+        # Fill-in the community bucket
+        bucket_id = self.data.bucket["id"]
+        self.data.community["bucket_id"] = bucket_id
+
+        yield Operation(OperationType.INSERT, FilesBucket, self.data.bucket)
+        yield Operation(OperationType.INSERT, Community, self.data.community)
+
+        oai_set_id = self.data.oai_set["id"]
+        self.data.oai_set["search_pattern"] = f"parent.communities.ids:{community_id}"
+        self.data.oai_set["system_created"] = True
+        yield Operation(OperationType.INSERT, OAISet, self.data.oai_set)
+
+        # Generate owner member
+        self.data.owner["community_id"] = community_id
+        yield Operation(OperationType.INSERT, CommunityMember, self.data.owner)
+
+        # Not every community has a logo
+        if self.data.community_file:
+            file_instance = self.data.file_instance
+            yield Operation(OperationType.INSERT, FilesInstance, file_instance)
+
+            object_version = self.data.object_version
+            logo_object_version_id = object_version["version_id"]
+            object_version["bucket_id"] = bucket_id
+            yield Operation(OperationType.INSERT, FilesObjectVersion, object_version)
+
+            community_file = self.data.community_file
+            community_file_id = community_file["id"]
+            community_file["record_id"] = community_id
+            community_file["object_version_id"] = logo_object_version_id
+            yield Operation(OperationType.INSERT, CommunityFile, community_file)
+
+        # Add to state
+        STATE.COMMUNITIES.add(
+            slug,
+            {
+                "id": community_id,
+                "bucket_id": bucket_id,
+                "oai_set_id": oai_set_id,
+                "community_file_id": community_file_id,
+                "logo_object_version_id": logo_object_version_id,
+            },
+        )
+
+
+@dataclass
+class CommunityUpdateData(LoadData):
+    """Community action data."""
+
+    community: dict
+
+    community_file: Optional[dict] = None
+    file_instance: Optional[dict] = None
+    object_version: Optional[dict] = None
+
+
+class CommunityUpdateAction(LoadAction):
+    """Update a community."""
+
+    name = "community-update"
+    data_cls = CommunityUpdateData
+
+    pks = [
+        # We generate a ComunityFile.id, just in case there was no logo before
+        ("community_file", "id", generate_uuid),
+    ]
+
+    def _generate_rows(self, **kwargs):
+        """Generates rows for a new community."""
+        slug = self.data.community["slug"]
```

Comment:
if it is a partial update, since the slug is not part of a pk, could it be that is not present in the payload?

Reply:
The issue is that in Zenodo legacy, the `slug` (which is `Community.id` in legacy) is the only identifier we have for the community, and I don't want to use the state in the transformer as per the discussion we had about delegating all PK/FK reference resolving to the Load.

I could be a bit more "optimistic" maybe about the input data and assume that if the `id` is in the community payload, we an skip resolving things. Though I have the feeling that in all the records/drafts streams we do a similar dereferencing using the `recid`.

Reply:
Damn was just reviewing your PRs and realized I should have put these in the `_resolve_references` to keep things a bit cleaner 🤦 . I'll do that once I finish reviewing your PRs.